src/templates/install.ps1

```diff
@@ -39,77 +80,59 @@ $role = $config.Role
 $environment = $config.Environment
 $isDelivery = $config.IsDelivery.ToLower() -eq "true"
 
-if(-not $websiteLocation) {
+if (-not $websiteLocation) {
     Write-Error "Website location is not set!"
 }
 
 if (-not $backupDir) {
     Write-Error "The backup dir is not set!"
 }
 
-if (-not $isDelivery)
-{
-    if(-not $itemsDirectory) {
+if (-not $isDelivery) {
+    if (-not $itemsDirectory) {
         Write-Error "itemsDirectory in config is mandatory"
     }
 }
 
-if ((Get-WebAppPoolState($appPoolName)).Value -ne "Stopped"){
-    Write-Output "Stopping IIS app pool $appPoolName"
-    (ls "IIS:\AppPools\$appPoolName\WorkerProcesses") | % {Get-Process -Id $_.processId | Stop-Process -Force}
-    Stop-WebAppPool $appPoolName
-}
-
-while((Get-WebAppPoolState($appPoolName)).Value -ne "Stopped") {
-    Write-Output "Application Pool $appPoolName not yet stopped."
-    sleep -Milliseconds 500
-}
-
-if(Test-Path $websiteLocation) {
-    Write-Output "Backup website at $websiteLocation to $backupDir"
-    Backup-WebRoot -Path $websiteLocation -BackupLocation $backupDir
-
-    Write-Output "Purge website location $websiteLocation"
-    rm "$websiteLocation\*" -Recurse -Force
-}
-else {
-    mkdir $websiteLocation
-}
+StopAppPool $appPoolName
 
-
-Write-Output "Copy content of $scriptPath\website to  $websiteLocation"
-cp "$scriptPath\website\*" "$websiteLocation\" -Recurse
+CopyWebsite "website" $websiteLocation $backupDir
 
 Install-LoftyEnvironment -WebRoot $websiteLocation `
     -Environment $environment `
     -Role $role `
     -BlueprintFolderPath $blueprintFolderPath `
     -ConfigFolder $configFolder
 
+StartAppPool $appPoolName
 
-Write-Output "Starting IIS app pool $appPoolName"
-Start-WebAppPool $appPoolName
+$additionalWebsites = $config.AdditionalWebsites.Website
```

Comment:
Is this really right here? Is it not only "$config.AdditionalWebsites" which have the array of websites?

Reply:
Yes, I tested this :)
$config.AdditionalWebsites is a single object, $config.AdditionalWebsites.Website is a list.